src/components/Store.ts

```diff
@@ -24,44 +35,115 @@ export class Store {
         return this.store
     }
 
-    getInstances() : { [key : string] : IInstance } {
-        return this.context.globalState.get<{
-            [key : string] : IInstance;
-        }>(InfluxDBInstancesKey) || {}
+    // Get the config. Don't cache this, otherwise we'd have to set up change
+    // event listeners.
+    get config() : vscode.WorkspaceConfiguration {
+        return vscode.workspace.getConfiguration('vsflux')
+    }
+
+    async getInstances() : Promise<{ [key : string] : IInstance }> {
+        const dataSource = this.config.get<string>('datasource', 'db')
```

Comment:
Nit: Add an enumeration for this

```typescript
enum DataSource {
    CLI = 'cli',
    DB = 'db',
}
```

Reply:
Yeah, I was a little inconsistent in this, because the enum is essentially a series of constants. Done.